src/adapter/src/catalog/state.rs

```diff
@@ -2418,6 +2434,35 @@ impl CatalogState {
                     Err(e) => panic!("unable to update system variable: {e:?}"),
                 }
             }
+            StateUpdateKind::Comment(comment) => match diff {
+                1 => {
+                    let prev = self.comments.update_comment(
+                        comment.object_id,
+                        comment.sub_component,
+                        Some(comment.comment),
+                    );
+                    soft_assert_eq_or_log!(
```

Comment:
This and the below soft_assert both seem like they should be panics because it means the catalog is in some inconsistent and unexpected state. Thoughts?

Reply:
Yeah that's a good point. I'll do that in a follow up PR, because there's actually a bunch of places like this.

Reply:
https://github.com/MaterializeInc/materialize/pull/26415